.github/workflows/ci.yml

```diff
@@ -92,3 +92,44 @@ jobs:
           DEFAULT_BRANCH: master
           # Not needed here as only one Linter is used.
           #GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  run-examples:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Install dependencies
+        run: |
+          sudo apt-get update;
+          DEBIAN_FRONTEND=noninteractive sudo apt-get install --no-install-recommends -yq \
+            jq \
+            libasound2-dev \
+            libudev-dev \
+            wget \
+            unzip \
+            xvfb;
+
+      - uses: actions/checkout@v2
+
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+
+      - name: Setup swiftshader
+        run: |
+          wget https://github.com/qarmin/gtk_library_store/releases/download/3.24.0/swiftshader.zip;
```

Comment:
Can we trust this? The repo doesn't have any readme or sources and doesn't tell how it was built. In addition there was exactly a single release. I think it would be better if we made a repo in the bevyengine org that uses CI to build swiftshader and download this pre-compiled version.

Reply:
origin from this download: https://github.com/godotengine/godot/issues/38428#issuecomment-808407836

I think we can trust it, but I agree it would be better to have a version owned by Bevy. I can setup a repo once Cart creates it

Reply:
Hmm as long as we aren't passing in secrets, do we need to trust it? Whats the worst case scenario here?

Reply:
Nothing really bad, mostly breaking our CI by changing/removing the download. Also we don't have the possibility to update if we need to

Reply:
I believe the github token passed for branch pushes (eg bors r+) has write access as far as I know. I don't know if it is possible to access it some way if it isn't explicitly used in the workflow file though.

Reply:
Yeah the github token definitely has write access. I _hope_ workflows only have access if it is explicitly passed in to a command in the workflow file. Otherwise we should reconsider our use of things like superlinter

Reply:
the ci for branch push is not triggered by Bors but by the event push on branch staging, I would consider it a big bug in GitHub Actions if Bors token leaked here

Reply:
You need write permission to push to a branch. If the github token is explicitly referenced in the workflow file it is definitively possible to use it to write things.

Reply:
I've dug into this a bit deeper. Everything I've read indicates that secrets are not accessible by a workflow container unless explicitly passed in. I don't have the bandwidth to do a full pen-test here, so I'll just trust what I've read (and the knowledge that if it didn't behave the way we've discussed, there would be very serious security implications that would completely erode people's trust in github).

Reply:
we discussed this a bit today, secrets are not accessible during pr tests, and the `GITHUB_TOKEN` should someone add it in an action would be a readonly token